lib/config.js

```diff
@@ -40,9 +40,14 @@ config.mongodb.authentication = {
   // used by MongoDB < 3.0
   // authMechanism: 'MONGODB-CR'
 };
-config.mongodb.connectOptions = {
+
+const writeConcern = config.mongodb.writeConcern = {
```

Comment:
This pattern does consolidate `writeConcern` in one place. We re-use `config.mongodb.writeConcern` everywhere we had those parameters before.

Reply:
I see them being used in `config.mongodb.connectOptions`, which could be edited independently from `config.mongodb.writeConcern`, resulting in the problem I'm worried about. I'm not talking about them being in the same variable in a file, I'm talking about exposing multiple interfaces (`config.mongodb.connectOptions`, `bedrock.mongodb.options`) that all use `writeConcern` but that could get out of sync with one another.

Consider the user that does this in an extending config file:

```js
config.mongodb.connectOptions.writeConcern = {/*something else*/};
```

This will cause `writeConcern` to get out of sync in the other places in a way that shouldn't be possible (devs get burned).

Can we either:

1. Ensure that we're using computed variables in some way such that if they are changed in one place, they are automatically changed in all the places, OR
2. Ensure that the code always "assembles" the config by overlaying `config.mongodb.writeConcern` on top of the other options where used, OR
3. Prevent changing `writeConcern` directly inside of `config.mongodb.connectOptions` (et. al), OR
4. Document the danger of changing it directly, OR
5. Some other option that mitigates the concern raised?

Reply:
This adds a layer of `writeConcern`, but does not freeze `writeConcern` itself: https://github.com/digitalbazaar/bedrock-mongodb/pull/65/commits/686656053ecc976ab6420dad2fe7f9af030106f8

Reply:
This is a proposal for a way of making `writeConcern` harder to mess up using an es6 class and getters and setters: https://gist.github.com/aljones15/a9197fc151d2e8ca39ad29de343c2dde